### DISCUSSION TITLE
Enhance OpenAPI Specification with Detailed Error Codes.

### DISCUSSION THREAD

**Discussion by ognis1205** [unitycatalog/unitycatalog#807] [Ideas]
**Is your feature request related to a problem? Please describe.**

To improve the UI/UX of client applications, such as the WebUI, appropriate error codes should be added to the OpenAPI specification. For instance, in the default implementation, attempting to delete a catalog that contains a schema is treated as an error. However, the response returned to the client provides no information about the reason for the failure, currently returning only a generic 400 Bad Request.

**Describe the solution you would like**

Add appropriate error codes to the OpenAPI specification and implement the corresponding error handling accordingly.

**Comment by ognis1205** [comment]
I think we should discuss this topic @dennyglee @MrPowers .

**Reply by dennyglee** [reply]
Thanks for your patience @ognis1205  - you got it!

**Comment by ognis1205** [comment]
Thank you for bringing this up for discussion @dennyglee!

I’d like to add a bit more context to this topic. The official Databricks UC API documentation does not include descriptions of error codes, so the current OpenAPI specification is in line with it. On the other hand, OSS Delta Sharing defines error codes in detail in its OpenAPI specification, even those not described in the official Delta Sharing API documentation:

 - https://docs.databricks.com/api/workspace/catalogs
 - https://github.com/delta-io/delta-sharing/blob/main/PROTOCOL.md

 This is particularly helpful for client development. For example, using a Prism mock server:
 
  - https://docs.stoplight.io/docs/prism/beeaad4dc0227-prism-cli#modifying-responses
 
  you can return specified error codes and responses. This allows development based on the OpenAPI specification without having to start an actual UC server, thus improving development efficiency.

**Reply by dennyglee** [reply]
Got it, would you be up for creating a PRD to suggest the flow of error codes/desc so we can start reviewing it?

**Reply by ognis1205** [reply]
Understood about the PRD as well. Summarizing the discussions from the other threads, I have to write two documents and share them with the community. Please give me a bit of time.

**Reply by ognis1205** [reply]
NOTE:

https://armeria.dev/docs/server-annotated-service/#handling-exceptions

**Reply by dennyglee** [reply]
Thanks like always @ognis1205 ! Please let me know (and tag me) when the PRD is ready so we can broadcast and perhaps bring it up for discussion in one of the future community meetups.  Thanks!

**Reply by ognis1205** [reply]
No problem at all @dennyglee ! I’ll continue preparing this alongside the WebUI refactoring proposal, so please wait a little longer. Since OpenAPI is one of the core aspects of the project, I’d love to hear everyone’s opinions, including ideas about its coding conventions.
